File: py/solved/41.py
def pancheck(x):
	s = str(x)
	l = len(s)
	for i in range(1,l+1):
		if str(i) not in s:
			return False
	return True

def prime(x):
	if x < 2:
		return False
	if x == 2:
		return True
	if x%2 == 0:
		return False
	for i in range(3,x,2):
		if x%i == 0:
			return False
	return True
def solve():
    N = 1000000000
    m = 0
    i = 1
    while i < N:
            if pancheck(i):
                    if prime(i):
                            m = i
            i += 1
    return m

from timer import time_function
print(time_function(solve))

File: cpp/solved/41.cpp
#include <iostream>
#include <chrono>
#include <vector>
#include "helper/prime.h"
#include "helper/stringfix.h"

using namespace std;

bool pandigital(int r, int n) {
    int pan [r+1];
    int i, tmp;
    for (i = 0; i < r+1; i++) {
        pan[i] = 0;
    }
    tmp = n;
    while (tmp > 0) {
        if (pan[tmp%10] == 1 || pan[tmp%10] > r) {
            return false;
        } else {
            pan[tmp%10] = 1;
        }
        tmp = tmp/10;
    }
    for (i = 1; i < r+1; i++) {
        if (pan[i] == 0) {
            return false;
        }
    }
    return pan[0] == 0;
}

int solve() {
    // 1-9 & 1-8 always divisible by 3
    long N = 10000000, n;
    for (n = N; n >= 0; n--) {
        if (prime(n) && pandigital(tostring(n).length(),n)) {
            break;
        }
    }
    return n;
}

int main() {
    typedef chrono::high_resolution_clock Clock;
    typedef chrono::milliseconds milliseconds;
    Clock::time_point start = Clock::now();
    cout << "Answer: " << solve() << endl;
    Clock::time_point end = Clock::now();
    milliseconds ms = chrono::duration_cast<milliseconds>(end-start);
    cout << "Time: " << ms.count() << " ms" << endl;
    return 0;
}
